apps/client/src/common/components/client-modal/RenameClientModal.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+import {
+  Button,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from '@chakra-ui/react';
+import { ClientList } from 'ontime-types';
+
+import style from './ClientModal.module.scss';
+
+interface RenameClientModalProps {
+  onClose: () => void;
+  isOpen: boolean;
+  id: string;
+  clients: ClientList;
+  onSubmit: (path: string) => void;
+}
+
+export function RenameClientModal(props: RenameClientModalProps) {
+  const { onClose, isOpen, id, clients, onSubmit } = props;
+  const [name, setName] = useState('');
+
+  // when clientlist and id is populated get the relevant name to use a default value
+  useEffect(() => {
+    if (clients[id]?.name) setName(clients[id].name);
+  }, [clients, id]);
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} variant='ontime'>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Rename {clients[id]?.name}</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <div>All connections in this client will have the same name after a reload</div>
```

Comment:
question: I am not sure this feedback is useful. could we rephrase or remove?

Reply:
Yeah maybe just note it some where in the docs